src/functTest/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPluginFunctionalTest.groovy

```diff
@@ -28,6 +29,68 @@ EXPOSE 8080
         result.output.contains('Author           : ')
     }
 
+    def "Can build image for Java application with default configuration and changes on changed application package"() {
+        String projectName = temporaryFolder.root.name
+        createJettyMainClass()
+        writeBasicSetupToBuildFile()
+        writeCustomTasksToBuildFile()
+
+        when:
+        BuildResult result = build('dockerBuildImage')
+
+        then:
+        File dockerfile = new File(projectDir, 'build/docker/Dockerfile')
+        File distTarFile = new File(projectDir, "build/distributions/${projectName}-1.0.tar")
+        distTarFile.exists()
+        long distTarFileLastTouched = distTarFile.lastModified()
+        dockerfile.exists()
+        dockerfile.text ==
+"""FROM java
+MAINTAINER ${System.getProperty('user.name')}
+ADD ${projectName}-1.0.tar /
+ENTRYPOINT ["/${projectName}-1.0/bin/${projectName}"]
+EXPOSE 8080
+"""
+
+        when:
+        result = build('dockerBuildImage')
+
+        then:
+        TaskOutcome.UP_TO_DATE == result.tasks.find{it.path == ":dockerBuildImage"}.outcome
```

Comment:
This fails in https://travis-ci.org/bmuschko/gradle-docker-plugin/jobs/265243034#L736
```
TaskOutcome.UP_TO_DATE == result.tasks.find{it.path == ":dockerBuildImage"}.outcome
                       |  |      |     |                                    |
                       |  |      |     :dockerBuildImage=SUCCESS            SUCCESS
                       |  |      [:compileJava=UP_TO_DATE, :processResources=NO_SOURCE, :classes=UP_TO_DATE, :jar=UP_TO_DATE, :startScripts=UP_TO_DATE, :distTar=UP_TO_DATE, :dockerCopyDistResources=UP_TO_DATE, :dockerDistTar=UP_TO_DATE, :dockerBuildImage=SUCCESS]
                       |  org.gradle.testkit.runner.internal.FeatureCheckBuildResult@68c9d179
                       false
```